lenskit-core/src/main/java/org/lenskit/data/entities/AbstractEntity.java

```diff
@@ -57,6 +57,16 @@ public EntityType getType() {
     /**
      * {@inheritDoc}
      *
+     * This implementation delegates to {@link #hasAttribute(String)}
+     */
+    @Override
+    public boolean hasAttribute(TypedName<?> name) {
+        return hasAttribute(name.getName());
```

Comment:
In practice, that shouldn't much matter, because either no-such-attribute or illegal-argument is the result of an entity failing to satisfy the contract required by the code using it.